docs/build_pipeline_controller.md

```diff
@@ -11,16 +11,23 @@ flowchart TD
 
   %% Node definitions
 predicate((PREDICATE: <br> Filter events related to <br> PipelineRuns <br> proccessed by Chains <br> that have <br> succeeded))
-get_pipeline_run{Pipeline found?}
+new_pipeline_run{Pipeline created?}
+get_pipeline_run{Pipeline updated?}
+finalizer_exists{Does the finalizer already exist?}
 retrieve_associated_entity(Retrieve the entity <br> component/application)
 determine_snapshot{Does a snapshot exist?}
 create_snapshot(Gather Application components<br> Add new component  <br> Create snapshot)
```

Comment:
we might also need a new node that describes the controller removing the finalizer as well.
Ideally after the nodes:
* retrieve_associated_entity --No  -->
* determine_snapshot         --Yes --> 
* (somewhere after) create_snapshot
* retrieve_associated_entity  --No -->

Reply:
Ack, mentioned above